src/library/required/extension/omg-client-logger-preloader.js

```diff
@@ -0,0 +1,110 @@
+(function (omg) {
+    omg = omg || {};
+
+    var CLIENT_ERROR_LABEL = 'omg.utag.client-error';
+    var POST_MESSAGE_COMMAND = 'client-log-error';
+    var SAFEFRAMES = 'safeframes';
+
+    var SafeFramesClientLogger = function () {};
+    SafeFramesClientLogger.prototype = {
+        logError: function (error, additionalData) {
+            if ('undefined' === typeof top || 'undefined' === top.postMessage) {
+                return;
+            }
+            additionalData = additionalData || [];
+            additionalData.push(SAFEFRAMES + '=true');
+            var errorMessage = this.createSafeframesMessage(error, additionalData);
+            var sourceUrl = '';
+            if ('undefined' !== typeof utag_data && 'undefined' !== typeof utag_data.parenturl) {
+                sourceUrl = utag_data.parenturl;
+            }
+            top.postMessage('cmd=' + POST_MESSAGE_COMMAND + '&value=' + JSON.stringify(errorMessage), sourceUrl);
+        },
+        /**
+         * @param error Javascript error object
+         * @param additionalData
+         * @returns {{message: string, fileName: string, stack: Array}}
+         */
+        createSafeframesMessage: function(error, additionalData) {
+            var msg = {
+                label: CLIENT_ERROR_LABEL,
+                message: "",
+                fileName: "",
+                stack: [],
+                additionalData: additionalData
+            };
+            msg.message = error.message;
+            msg.fileName = getSource(error);
+            msg.stack = parseStack(error);
+            return msg;
+        }
+    };
+
+    var DctkClientLogger = function () {};
+    DctkClientLogger.prototype = {
+        logError: function (error, additionalData) {
+            additionalData = additionalData || [];
+            additionalData.push(SAFEFRAMES + '=false');
+            error.fileName = getSource(error);
+            error.stack = parseStack(error).join(", ");
+            dctk.logging.logError(CLIENT_ERROR_LABEL, error, additionalData);
+        }
+    };
+
+    var NoOpClientLogger = function () {};
+    NoOpClientLogger.prototype = {
+        logError: function (error, additionalData) {}
+    };
+
+    /**
+     * @param error Javascript error Object
+     * @returns The page source or the entire stack string.
+     */
+    function getSource(error) {
+        if ('string' === typeof error.stack) {
+            var stackArr = error.stack.split(/[\r\n]+/g);
+            if (stackArr.length > 1) {
+                return parseSource(stackArr[1]);
+            }
+        }
+        return error.stack;
+    }
+
+    function parseSource(rawLine) {
+        var line = rawLine.replace(/at/gi, '').trim();
+        if (line.indexOf('?') !== -1) {
```

Comment:
can't remember, but don't think you need this check. if there's no '?', it should just return the original string.